### DISCUSSION TITLE
[Request] Disable water safe teleportation

### DISCUSSION THREAD

**Discussion by TheWranger** [Zrips/CMI#1088] [Enhancement]
**Description of issue or feature request:**
Would really like the idea to block people from being teleported into water via safe teleportation. Can be a bit of an annoyance a times for players and would like to be able to block it as a server owner.
Thanks.

**Comment by LogGits** [comment]
Are you talking about the random teleport feature or just block `/tpa`'s to players who are in water?

**Comment by TheWranger** [comment]
I mean like when you get moved to a safe location

**Comment by LogGits** [comment]
Ah ok, have you tried this on build 8.2.0.0? There is a new teleportation system. I recommend trying this with `SafeLocationDownThenUp: true` and seeing if it fixes your issue, if not, don't hesitate to respond here 🙂

**Comment by TheWranger** [comment]
Haven't updated to 8.2.0.0 so I'll give that a try, but am I correct is saying that people will still sometimes get teleported into water from the new version? Was hoping to completely block safe teleportation into water (as if it was lava).

**Comment by LogGits** [comment]
Ye water is usually safe for teleportation. Just out of curiosity, what happens if a player dies in an ocean biome, where shall we tp them if it's only water. The tp system was dramatically improved in 8.2.0.0 and is probably one of the best teleportation systems in any plugin, being able to teleport into intricate stair structures with ease. You will see less "Teleporting you to a safe place" because if a player can stand in a location they can teleport there.

**Comment by TheWranger** [comment]
Alright, I more just wanted it added as a config option for servers that might want to stop people from getting teleported into water if they need it for whatever gamemode they are running and know what other issues it might cause.

**Comment by LogGits** [comment]
Ah ok i can see how that would be useful to have 👍
